src/net_processing.cpp

```diff
@@ -760,6 +764,12 @@ void PeerLogicValidation::NewPoWValidBlock(const CBlockIndex *pindex, const std:
     bool fWitnessEnabled = IsWitnessEnabled(pindex->pprev, Params().GetConsensus());
     uint256 hashBlock(pblock->GetHash());
 
+    {
```

Comment:
I'm confused as to exactly what you're suggesting here. I don't particularly want a weak_ptr because I do want to hold the latest block for responses for a while after the block is connected/broadcast.